src/picongpu/include/fields/FieldJ.kernel

```diff
@@ -361,83 +361,145 @@ struct KernelAddCurrentToEMF
     }
 };
 
+template< uint32_t T_workers >
 struct KernelBashCurrent
 {
-    template<class Mapping>
-    DINLINE void operator()(J_DataBox fieldJ,
-                                      J_DataBox targetJ,
-                                      DataSpace<simDim> exchangeSize,
-                                      DataSpace<simDim> direction,
-                                      Mapping mapper) const
+    template< typename Mapping >
+    DINLINE void operator()(
+        J_DataBox const & fieldJ,
```

Comment:
You are right it is illegal in a CUDA `__global__` function.
In this case it fine to use the reference because the functor is already within a kernel.